File: 06_Sum_Square_Difference/C/06.c
#include <stdio.h>

unsigned long sum_of_squares(unsigned short limit) {
  unsigned long sum = 0;
  for(unsigned short i = 1; i <= limit; i++) {
    sum += i * i;
  }
  return sum;
}

unsigned long square_of_sum(unsigned short limit) {
  unsigned long sum = 0;
  for(unsigned short i = 1; i <= limit; i++) {
    sum += i;
  }
  sum *= sum;
  return sum;
}

int main() {
  unsigned short limit = 100;
  printf("%lu\n", square_of_sum(limit) - sum_of_squares(limit));
  return 0;
}

File: 06_Sum_Square_Difference/Rust/06.rs
fn sum_of_many(limit: i32) -> i32 {
    let mut sum_of_squares = 0;
    let mut square_of_sum = 0;

    for num in 1..=limit {
        sum_of_squares += num * num;
        square_of_sum += num;
    }
    println!("{sum_of_squares}");
    println!("{square_of_sum}");
    let result = (sum_of_squares - square_of_sum * square_of_sum).abs();
    return result;
}

fn main() {
    println!("{}", sum_of_many(100));
}